Awful.apk/src/main/java/com/ferg/awfulapp/ProgressListener.kt

```diff
@@ -0,0 +1,16 @@
+package com.ferg.awfulapp
```

Comment:
Does this need to be in its own file? It's only used by ``AwfulRequest`` to talk to the AwfulFragment superclasses, it feels like it makes more sense staying in there

Reply:
Yes.... Unless you know the solution to the weird generics issue it causes. Since AwfulRequest takes a generic type, when converting this code to Kotlin, the overrides in AwfulFragment failed to be recognized as proper overrides, and so every subclass failed to compile saying that it needs to implement the members. I'd love another opinion on this. I like to think I know Kotlin pretty well, but generics are an issue that I still struggle with.

Reply:
Oh I see what you mean... it's the Java -> Kotlin -> Java thing with the inheritance, right?

I just did this to the listener
````
    public interface ProgressListener<T>{
        void requestStarted(AwfulRequest<T> req);
        void requestUpdate(AwfulRequest<T> req, int percent);
        void requestEnded(AwfulRequest<T> req, VolleyError error);
    }
````
and then set the implementation types to ``AwfulRequest<Any>`` and that shut it up. I guess the raw types cause a problem and we shouldn't really be using them anyway, so this fix is fine by me!